### PR TITLE
OAK-11305 : use single delete call for renameKeys() in GCP

### PR DESCRIPTION
**Pull request by t-rana**

### CODE REVIEW

File: oak-blob-cloud/src/main/java/org/apache/jackrabbit/oak/blob/cloud/s3/S3Backend.java [hunk 1]
```diff
@@ -1229,22 +1229,24 @@ private void renameKeys() throws DataStoreException {
         long count = 0;
         try {
             Thread.currentThread().setContextClassLoader(
-                getClass().getClassLoader());
+                    getClass().getClassLoader());
```
Comment (rishabhdaim) [inline oak-blob-cloud/src/main/java/org/apache/jackrabbit/oak/blob/cloud/s3/S3Backend.java]: please revert this.

File: oak-blob-cloud/src/main/java/org/apache/jackrabbit/oak/blob/cloud/s3/S3Backend.java [hunk 2]
```diff
@@ -1229,22 +1229,24 @@ private void renameKeys() throws DataStoreException {
...
             ObjectListing prevObjectListing = s3service.listObjects(bucket);
             List<DeleteObjectsRequest.KeyVersion> deleteList = new ArrayList<DeleteObjectsRequest.KeyVersion>();
+            List<String> keysToDelete = new ArrayList<>();
             int nThreads = Integer.parseInt(properties.getProperty("maxConnections"));
             ExecutorService executor = Executors.newFixedThreadPool(nThreads,
-                new NamedThreadFactory("s3-object-rename-worker"));
+                    new NamedThreadFactory("s3-object-rename-worker"));
```
Comment (rishabhdaim) [inline oak-blob-cloud/src/main/java/org/apache/jackrabbit/oak/blob/cloud/s3/S3Backend.java]: same as above

**Review by rishabhdaim — CHANGES_REQUESTED** [review]

File: oak-blob-cloud/src/main/java/org/apache/jackrabbit/oak/blob/cloud/s3/S3Backend.java [hunk 3]
```diff
@@ -1257,34 +1259,39 @@ private void renameKeys() throws DataStoreException {
...
-                    && !executor.awaitTermination(10, TimeUnit.SECONDS)) {
+                        && !executor.awaitTermination(10, TimeUnit.SECONDS)) {
                     LOG.info("Rename S3 keys tasks timedout. Waiting again");
                 }
             } catch (InterruptedException ie) {
 
             }
             LOG.info("Renamed [{}] keys, time taken [{}]sec", count,
-                ((System.currentTimeMillis() - startTime) / 1000));
+                    ((System.currentTimeMillis() - startTime) / 1000));
```
Comment (rishabhdaim) [inline oak-blob-cloud/src/main/java/org/apache/jackrabbit/oak/blob/cloud/s3/S3Backend.java]: same as above

**Comment by t-rana** [comment]
removed all format changes.

**Review by rishabhdaim — APPROVED** [review]

**Review by rishabhdaim — APPROVED** [review]

**Pull request merged**